core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -736,6 +748,12 @@ public void updateVersion(long version) {
             this.doc.version().setLongValue(version);
         }
 
+        @Override
+        public void updateSeqNo(long seqNo) {
+            super.updateSeqNo(seqNo);
+            this.doc.seqNo().setLongValue(seqNo);
```

Comment:
I haven't looked yet but why is this mutable? should we maybe return a new Instance from the engine and make this immutable?